### PR TITLE
docs(container): additional context and app overlay

### PR DESCRIPTION
**Pull request by hannachen**
This is linked from our public docs, updated with information about containers and app overlay.

### CODE REVIEW

**Comment by ow** [comment]
Looks good to me! I made a tiny clarifying commit but otherwise :)

**Review by ow — APPROVED** [review]

File: docs/Utilities/Container.md [hunk 1]
```diff
@@ -1,8 +1,20 @@
...
-## Subscription Management
+## Product Subscription
+
+There are two different container types for the Product subscription extension, depending on the extension mode:
+
+#### Add, Remove
+**App overlay container**: A full-screen container designed for complex tasks, like configuring a subscription for the first time.
+
+#### Create, Edit
+**App modal container**: A smaller overlay typically used for editing an existing configuration or confirming a change. In the app modal container, primary and secondary action buttons are responsible for triggering a custom behaviour, such as performing a resource update.
```
Comment (dave-pelletier) [inline docs/Utilities/Container.md:13]: These are backwards

File: docs/Utilities/Container.md [hunk 2]
```diff
@@ -1,8 +1,20 @@
...
+
+#### Add, Remove
+**App modal container**: A smaller overlay typically used for editing an existing configuration or confirming a change. In the app modal container, primary and secondary action buttons are responsible for triggering a custom behaviour, such as performing a resource update.
+
+#### Create, Edit
+**App overlay container**: A full-screen container designed for complex tasks, like configuring a subscription for the first time.
+
+### Container API
+
+#### Add, Remove (Modal container)
```
Comment (alanthai) [inline docs/Utilities/Container.md:17]: I think this info should be found in Product Subscription page and all the relevant information about it is collocated as to not bloat the Container.md and Data.md with specific implemtnation in the future.

**Comment by hannachen** [comment]
Holding off additional work/merging this until @alanthai is complete with his PR: https://github.com/Shopify/app-extension-libs/pull/992 We can either close this after moving any relevant info or update to match the final docs in https://github.com/Shopify/app-extension-libs

**Comment by hannachen** [comment]
Closing this as the contents here have been moved to [Alan's PR](https://github.com/Shopify/argo-admin-template/pull/74/files), thanks for picking this up!

**Pull request closed without merging**